### DISCUSSION TITLE
Typing of Prop in Custom Hit Component

### DISCUSSION THREAD

**Discussion by theovier** [algolia/instantsearch#5970] [Q&A] [answered]
Hi,

I'm currently trying to correctly type my `Hit` and I'm a bit lost.

Assuming I have this setup

```ts
<InstantSearch indexName="books" searchClient={searchClient}>
     <SearchBox />
     <Hits hitComponent={Hit} />
</InstantSearch>
```

Further assuming I know that the hit documents will include an attribute called `name`. How do I type the prop of my `Hit` component correctly? I tried the following which is obviously not working:

```ts
const Hit: React.FC<{ hit: HitsProps<{ name: string }> }> = ({ hit }) => {
    return <div>{hit.name}</div>; // TS2339: Property  name  does not exist on type  HitsProps<{ name: string; }>
};
```

How do I type the hit correctly so that I can access my document's attributes typesafe in my custom `Hit` component?

Thanks! :)

**Comment by dhayab** [comment, marked as answer]
Here's an example of how it could be typed:

```tsx
import type { Hit } from 'instantsearch.js';

type HitProps = {
  hit: Hit<{
    name: string;
  }>;
};

const Hit: React.FC<HitProps> = ({ hit }) => {
  return <div>{hit.name}</div>;
};
```

**Reply by theovier** [reply]
Ah, I missed the `Hit` type. Thanks!
